pkg/operator/ceph/csi/operator_config.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2024 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"reflect"
+
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+	opcontroller "github.com/rook/rook/pkg/operator/ceph/controller"
+
+	csiopv1a1 "github.com/ceph/ceph-csi-operator/api/v1alpha1"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	k8scsiv1 "k8s.io/api/storage/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const opConfigCRName = "ceph-csi-operator-config"
+
+func (r *ReconcileCSI) createOrUpdateOperatorConfig(cluster cephv1.CephCluster, clusterInfo *cephclient.ClusterInfo) error {
+	logger.Info("Creating ceph-CSI operator config CR")
+
+	opConfig := &csiopv1a1.OperatorConfig{}
+	opConfig.Name = opConfigCRName
+	opConfig.Namespace = r.opConfig.OperatorNamespace
+
+	imageSetCmName, err := r.createImageSetConfigmap(clusterInfo)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create ceph-CSI operator config ImageSetConfigmap for CR %s", opConfigCRName)
+	}
+
+	spec := r.generateCSIOpConfigSpec(cluster, opConfig, imageSetCmName)
+	opConfig.Spec = spec
+
+	err = r.client.Get(r.opManagerContext, types.NamespacedName{Name: opConfigCRName, Namespace: r.opConfig.OperatorNamespace}, opConfig)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			err = r.client.Create(r.opManagerContext, opConfig)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create CSI-operator operator config CR %q", opConfig.Name)
+			}
+
+			logger.Info("Successfully created ceph-CSI operator config CR")
+			return nil
+		}
+		return errors.Wrapf(err, "failed to get CSI-operator operator config CR %q", opConfigCRName)
+	}
+
+	err = r.client.Update(r.opManagerContext, opConfig)
+	if err != nil {
+		return errors.Wrapf(err, "failed to update CSI-operator operator config CR %q", opConfig.Name)
+	}
+	logger.Info("Successfully updated ceph-CSI operator config CR")
+
+	return nil
+}
+
+func (r *ReconcileCSI) generateCSIOpConfigSpec(cluster cephv1.CephCluster, opConfig *csiopv1a1.OperatorConfig, imageSetCmName string) csiopv1a1.OperatorConfigSpec {
+	maxLogSize, _ := opcontroller.GetLogRotateConfig(cluster.Spec)
```

Comment:
log rotation is on hold for a bit right now, I'd suggest removing it from this PR.

Reply:
removed